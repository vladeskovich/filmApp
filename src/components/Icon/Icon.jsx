import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Star from '../../assets/icons/star.svg';
import Arrow from '../../assets/icons/arrow.svg';
import List from '../../assets/icons/list.svg';
import Grid from '../../assets/icons/grid.svg';
import Play from '../../assets/icons/play.svg';
import Cross from '../../assets/icons/cross.svg';

import styles from './Icon.scss';

const icons = {
  play: Play,
  list: List,
  grid: Grid,
  cross: Cross,
  arrow: Arrow,
  star: Star,
};

const Icon = ({
  name,
  className,
  onClick,
  ...props
}) => {
  const classes = classNames(styles.icon, className);
  const Component = icons[name.toLowerCase()];

  return (
    <Component
      onClick={onClick}
      className={classes}
      {...props}
    />
  );
};

Icon.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  onClick: PropTypes.func,
};

Icon.defaultProps = {
  name: 'Cross',
};

export default Icon;
