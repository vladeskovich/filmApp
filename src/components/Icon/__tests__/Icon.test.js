import React from 'react';
import Icon from '../Icon';

describe('Icon', () => {
  const props = {
    name: 'Star',
    onClick: () => {},
    className: 'buttonTest',
  };

  it('should render Icon component', () => {
    const component = mount(<Icon/>);

    expect(component).toMatchSnapshot();
  });

  it('should render Icon component with props', () => {
    const component = mount(<Icon {...props}/>);

    expect(component).toMatchSnapshot();
  });
});
