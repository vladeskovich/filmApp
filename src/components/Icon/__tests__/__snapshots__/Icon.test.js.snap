// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render Icon component 1`] = `
<Icon
  name="Cross"
>
  <SvgCross
    className="Icon__icon_2eyT0_yM"
  >
    <svg
      className="Icon__icon_2eyT0_yM"
      data-file-name="SvgCross"
    />
  </SvgCross>
</Icon>
`;

exports[`Icon should render Icon component with props 1`] = `
<Icon
  className="buttonTest"
  name="Star"
  onClick={[Function]}
>
  <SvgStar
    className="Icon__icon_2eyT0_yM buttonTest"
    onClick={[Function]}
  >
    <svg
      className="Icon__icon_2eyT0_yM buttonTest"
      data-file-name="SvgStar"
      onClick={[Function]}
    />
  </SvgStar>
</Icon>
`;
